src/Microsoft.IdentityModel.JsonWebTokens/Json/JsonClaimSet.cs

```diff
@@ -24,14 +25,44 @@ internal class JsonClaimSet
 
         internal object _claimsLock = new();
         internal readonly Dictionary<string, object> _jsonClaims;
+
+#if NET8_0_OR_GREATER
+        internal readonly Dictionary<string, (int startIndex, int length)?> _jsonClaimsBytes;
+        internal readonly Memory<byte> _tokenAsMemory;
```

Comment:
Should this be mutable?
```suggestion
        internal readonly ReadOnlyMemory<byte> _tokenAsMemory;
```

Reply:
Needs to be Memory since we're writing into it when unescaping.